### PR TITLE
fix(RouteViewer): Fetch positions in route and pattern views.

### LINKED ISSUE #847: Vehicle positions no longer fetched in route viewer
When clicking a transit route in the route viewer, the route shape is drawn on the map, however vehicle positions for that route are no longer fetched and should be fetched.
This issue occurs as of #829.

### PR DESCRIPTION
**Pull request by binh-dam-ibigroup**
## Description

Fix #847 using new component `VehiclePositionRetriever`.

## PR Checklist

- [na] Does the code follow accessibility standards (WCAG 2.1 AA Compliant)?
- [na] Are all languages supported (Internationalization/Localization)?
- [x] Are appropriate Typescript types implemented?

### CODE REVIEW

File: lib/components/viewers/vehicle-position-retriever.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    }
+  }, [routeId, getVehiclePositionsForRoute])
+
+  useEffect(() => {
+    // Fetch vehicle positions when initially mounting component and a route id is available.
+    if (routeId) {
+      refreshVehiclePositions()
+
+      if (!refreshTimer) {
+        console.log('Set up timer for route ' + routeId || '')
```
Comment (amy-corson-ibigroup) [inline lib/components/viewers/vehicle-position-retriever.ts:34]: Can we change the verb here to match "unmounting" below?
```suggestion
        console.log('Setting up timer for route ' + routeId || '')
```

**Review by amy-corson-ibigroup — COMMENTED** [review]
Very cool fix! Does this work in tandem with the `refreshVehiclePositions` method in `RouteDetails` or can that be removed/replaced by this component?

File: lib/components/viewers/vehicle-position-retriever.ts (hunk 1 above)
Comment (binh-dam-ibigroup) [reply lib/components/viewers/vehicle-position-retriever.ts:34]: Actually I forgot to remove these log statements!

**Comment by binh-dam-ibigroup** [comment]
> Very cool fix! Does this work in tandem with the `refreshVehiclePositions` method in `RouteDetails` or can that be removed/replaced by this component?

How did I forget to delete the existing code in `RouteDetails`...

File: lib/components/viewers/vehicle-position-retriever.ts (hunk 1 above)
Comment (binh-dam-ibigroup) [reply lib/components/viewers/vehicle-position-retriever.ts]: Fixed in https://github.com/opentripplanner/otp-react-redux/pull/848/commits/16c0707acde81511e8ef7aa4da4e4919869d76a0.

**Comment by binh-dam-ibigroup** [comment]
> > Very cool fix! Does this work in tandem with the `refreshVehiclePositions` method in `RouteDetails` or can that be removed/replaced by this component?
> 
> How did I forget to delete the existing code in `RouteDetails`...

I removed that old code in https://github.com/opentripplanner/otp-react-redux/pull/848/commits/16c0707acde81511e8ef7aa4da4e4919869d76a0.

**Review by amy-corson-ibigroup — APPROVED** [review]
Looks great!!

**Review by AdrianaCeric — APPROVED** [review]
Great work!

File: lib/components/viewers/vehicle-position-retriever.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+      }
+    }
+  }, [routeId, refreshVehiclePositions, refreshTimer, refreshSeconds])
+
+  // Component renders nothing.
+  return null
+}
+
+// connect to redux store
+const mapStateToProps = (state: any) => {
```
Comment (AdrianaCeric) [inline lib/components/viewers/vehicle-position-retriever.ts:55]:
```suggestion
const mapStateToProps = (state: Record<string, any>) => {
```

**Pull request merged**